app/pages/lab-pages-editor/helpers/linkStepsInWorkflow.js

```diff
@@ -0,0 +1,54 @@
+/*
+Links Steps in a Workflow together.
+
+- Input: array of Steps, and Tasks object.
+- Output: COPY of array of Steps, with their `step.next`s updated.
+
+Rules for linking steps:
+- If a Step *can* branch, then step.next = undefined.
+- If a Step is the final step in the array, then step.next = undefined. 
+- Otherwise, each Step's step.next will be the ID of the next Step in the array.
+ */
+
+export default function linkStepsInWorkflow(steps = [], tasks = {}) {
+  const newSteps = steps.map((step, index) => {
+    const [stepId, stepBody] = step;
+
+    const isFinalStep = index >= (steps.length - 1);
```

Comment:
This is true for linear workflows, which don't need `step.next` anyway, but branched workflows will have more than one final step (one final step for the end of each branch.)

Reply:
In this example, `S0` branches to either `S1` or `S2`, both of which are final steps ie. you can submit a classification either from `S1` or from `S2`.

https://github.com/zooniverse/front-end-monorepo/blob/45dde3a406671d480e97142789375d595afa086a/packages/lib-classifier/src/store/helpers/convertWorkflowToUseSteps/convertWorkflowToUseSteps.spec.js#L243-L259

Reply:
Dangit, this variable name was more confusing than I realised. I'll rename it to `isFinalStepInArray` 👌 

> If a Step is the final step in the array, then step.next = undefined.

Reply:
Can the final step in the array loop back to an earlier step, for recursive workflows? An example of this would be transcribing tables with a variable number of entries (Scarlets & Blues.)

Reply:
Yes, that's possible, but the project owner would need to (1) make the final Step/Page (in the array) a Single Question task to routes users back to an earlier step, and (2) manually create an off-ramp at some earlier step (which would also be a Single Question Task)

Example 1:

<img width="391" alt="image" src="https://github.com/zooniverse/Panoptes-Front-End/assets/13952701/a556815e-21d6-4a3b-b233-11f503c9424b">

Example 2: (trimmed)

<img width="389" alt="image" src="https://github.com/zooniverse/Panoptes-Front-End/assets/13952701/e7344aa9-31ed-46eb-8184-ee9fcab5c58a">

<img width="381" alt="image" src="https://github.com/zooniverse/Panoptes-Front-End/assets/13952701/ec7c6c05-e75b-4fee-b29e-17755796f42d">

I believe the goal for this new Pages Editor design is to encourage project owners to make the workflow _as linear as possible,_ by default.

Reply:
Is that second example Scarlets & Blues? That’s exactly the workflow that I had in mind.